pkg/compactor/blocks_cleaner.go

```diff
@@ -493,3 +522,21 @@ func listBlocksOutsideRetentionPeriod(idx *bucketindex.Index, threshold time.Tim
 
 	return
 }
+
+// findLastModifiedTimeForBlock finds the most recent modification time for all files in a block
+func findLastModifiedTimeForBlock(ctx context.Context, blockID ulid.ULID, userBucket objstore.InstrumentedBucket, userLogger log.Logger) (time.Time, error) {
```

Comment:
"last" in the name of function is not clear (is it oldest? or most recent?). Let's be explicit: `findMostRecentModifiedTimeForBlock`.